src/set_theory/cardinal/ordinal.lean

```diff
@@ -740,6 +740,27 @@ begin
   rintro ⟨t, h1t, h2t⟩, exact le_trans (mk_preimage_of_injective _ _ subtype.val_injective) h2t
 end
 
+@[simp] lemma mk_subset_interval {α : Type u} (r : α → α → Prop) [is_well_order α r] (a : α) :
+  #{s : set α // ∀ b ∈ s, r b a} = 2 ^ (ordinal.typein r a).card :=
```

Comment:
A better version of this statement might be
```suggestion
  #{b : α // r b a} = (ordinal.typein r a).card :=
```
Along with a lemma that says
```
#{s : set α // ∀ b ∈ s, p b} = #(set (subtype p))
```

Reply:
The former already exists, it is `card_typein`. There's no point in the latter, since the simp-normal form of `#(set a)` is `2 ^ (#a)`.

Reply:
It's ok (and often better) for the RHS to not be simp-normal; it's the LHS that has to not be.

If you feel strongly, then yes, you could instead state the lemma as
```lean
#{s : set α // ∀ b ∈ s, p b} = 2 ^ #(subtype p)
```

Reply:
Oh, I see. It might be best to make this a lemma about the number of subsets of a set. Seems simple enough to make into an equivalence.

Reply:
I think the equivalence is defeq to https://leanprover-community.github.io/mathlib_docs/logic/equiv/basic.html#equiv.subtype_pi_equiv_pi

Reply:
That can't be the case, since the RHS of that lemma is a function and not a subtype.

Reply:
Replace all the pi types with sets in that definition

Reply:
I managed to get the equivalence `{s : set α // s ⊆ t} ≃ Π (a : α), {h : Prop // h → t a}` out of this lemma, not sure how to show that the RHS is equivalent to `set t`.

Reply:
`{h : Prop // h → t a}` is the same as `t a

Reply:
I've asked for help on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Set.20of.20subsets.20of.20.60t.60.20is.20equivalent.20to.20.60set.20t.60/near/284299106).